domain/infosync/resource_group_manager.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infosync
+
+import (
+	"context"
+	"sync"
+
+	"github.com/pingcap/errors"
+	rmpb "github.com/pingcap/kvproto/pkg/resource_manager"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/parser/model"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// ConvertToProtoResourceGroup converts model.ResourceGroupInfo to rmpb.ResourceGroup
+func ConvertToProtoResourceGroup(modelGroup *model.ResourceGroupInfo) (*rmpb.ResourceGroup, error) {
+	group := &rmpb.ResourceGroup{
+		Name:     modelGroup.Name.String(),
+		Settings: &rmpb.GroupSettings{},
+	}
+	var isRUMode bool
+	if modelGroup.RRURate > 0 || modelGroup.WRURate > 0 {
```

Comment:
So, we need to say that RU mode can override native mode. We need to say that in our spec and the release ontes.

Reply:
What about let the parser only accept either RU mode or Naive mode but not both, then we can avoid these extra checks

Reply:
And, is it rational to support both RU and native mode resource groups for a single cluster at the same time?

Reply:
Not supports at same time. I will do more checks after this PR.